### ISSUE TITLE
[ISO9660] PS2 ISO does not extract properly extract, "Stream was too long"

### ISSUE THREAD

**Issue by Whovian9369** [claunia/DiscImageChef#678]
### Prerequisites

* [x] Are you running the latest version?
* [x] Can you reproduce the problem in the debug version?

### Check and fill as appropriate:
* [x] I was running Aaru under Linux, built via `aur/aaru` using `makepkg -si` - `aaru 5.3.0+caab6cde built by whovian-build-box in Release, modified Copyright © 2011-2021 Natalia Portillo`
* [x] ...in 64-bit
* [x] ...using an Intel / AMD processor
    * **Image format**:  ISO
    * **Application that created the image**: DiscImageCreator
* [x] I was extracting the filesystem contents...
    * ...**Filesystem**: ISO9660
    * [x] `fs info` command recognizes the filesystem

### Description

When trying to extract Redump PS2 image `Sonic Mega Collection Plus (USA) (Beta)`, it extracts some files before "completing" with `Error reading file: Stream was too long.`
Only around 4 MB of data is extracted, which is the main executable and (possibly incomplete?) Modules folder.

### Exact command line used:

`aaru fs extract "Sonic Mega Collection Plus (USA) (Beta).iso" "Sonic Mega Collection Plus (USA) (Beta)"`

### Expected behavior:
Full image filesystem extracted without erroring

### Actual behavior:
Error occurs after some data is extracted.

### Output of command without debug output enabled
```
aaru 5.3.0+caab6cde built by whovian-build-box in Release, modified
Copyright © 2011-2021 Natalia Portillo

Image format identified by Raw Disk Image.
1 partitions found.

Partition 0:
Identifying filesystem on partition
Identified by 2 plugins
As identified by ISO9660 Filesystem.
Written 57 bytes of file SYSTEM.CNF to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/SYSTEM.CNF
Written 3016400 bytes of file SLUS_209.17 to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/SLUS_209.17
Created subdirectory at Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES
Written 58037 bytes of file CRI_ADXI.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/CRI_ADXI.IRX
Written 15653 bytes of file DBCMAN.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/DBCMAN.IRX
Written 10469 bytes of file DS1O_D.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/DS1O_D.IRX
Written 11725 bytes of file DS2U_D.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/DS2U_D.IRX
Written 10497 bytes of file GGSOUND.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/GGSOUND.IRX
Written 275345 bytes of file IOPRP300.IMG to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/IOPRP300.IMG
Written 28661 bytes of file LIBSD.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/LIBSD.IRX
Written 20221 bytes of file MC2_D.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/MC2_D.IRX
Written 28717 bytes of file MSIS_IOP.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/MSIS_IOP.IRX
Written 9161 bytes of file SDRDRV.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/SDRDRV.IRX
Written 11289 bytes of file SIO2D.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/SIO2D.IRX
Written 6641 bytes of file SIO2MAN.IRX to Sonic Mega Collection Plus (USA) (Beta)/ISO9660/NO NAME/MODULES/SIO2MAN.IRX
Error reading file: Stream was too long.
```

### Output of command execution with debug output enabled
Section intentionally left blank

### Other comments
* Image is ~4.5 GB and the SSH connection I was using seems to have been destroying my internet connection when running with `-d`, so no log is provided. Contact me on Discord if you would indeed like a log, please.

**Comment by claunia** [comment]
Please attach output of `aaru fs ls "Sonic Mega Collection Plus (USA) (Beta).iso"`

**Comment by Whovian9369** [comment]
```
$ aaru fs ls "Sonic Mega Collection Plus (USA) (Beta).iso"
aaru 5.3.0+caab6cde built by whovian-build-box in Release, modified
Copyright © 2011-2021 Natalia Portillo

Image format identified by Raw Disk Image.
1 partitions found.

Partition 0:
Identifying filesystem on partition
Identified by 2 plugins
As identified by ISO9660 Filesystem.
Root directory
                         <DIR>                 MODULES
                         <DIR>                 MDROM_EU
                         <DIR>                 MDROM_US
                         <DIR>                 PAE_E
                         <DIR>                 PAE_F
                         <DIR>                 PAE_G
                         <DIR>                 PAE_I
                         <DIR>                 PAE_J
                         <DIR>                 PAE_K
                         <DIR>                 PAE_S
                         <DIR>                 TEXT
                         <DIR>                 PAE_E_EU
                         <DIR>                 COMIC
                         <DIR>                 COMMON
                         <DIR>                 E3
                         <DIR>                 GGROM
                         <DIR>                 ILLUST
                         <DIR>                 MANUAL
                         <DIR>                 MANUAL_E
                         <DIR>                 MDROM
                         <DIR>                 MODE_GS
                         <DIR>                 SOUND
                         <DIR>                 MOVIE
                            580            57  SYSTEM.CNF
                            581     3,016,400  SLUS_209.17
                           2298 2,954,080,256  _DUMMY_.BIN

Directory: MODULES
                           2054        58,037  CRI_ADXI.IRX
                           2083        15,653  DBCMAN.IRX
                           2091        10,469  DS1O_D.IRX
                           2097        11,725  DS2U_D.IRX
                           2103        10,497  GGSOUND.IRX
                           2109       275,345  IOPRP300.IMG
                           2244        28,661  LIBSD.IRX
                           2258        20,221  MC2_D.IRX
                           2268        28,717  MSIS_IOP.IRX
                           2283         9,161  SDRDRV.IRX
                           2288        11,289  SIO2D.IRX
                           2294         6,641  SIO2MAN.IRX

Directory: MDROM_EU
                         1444720     1,782,197  COMIX.CPT
                         1445591        46,934  FLICKY.CPT
                         1445614     1,886,551  KNU_P1.CPT
                         1446536     2,615,208  KNU_P2.CPT
                         1447813     2,866,160  KNU_P3.CPT
                         1449213       589,801  OOZE.CPT
                         1449501       641,565  PUYO_USA.CPT
                         1449815     1,575,127  RISTER.CPT
                         1450585     1,778,496  SANDK.CPT
                         1451454       423,591  SONIC1.CPT
                         1451661       425,672  SONIC1B.CPT
                         1451869       425,672  SONIC1U.CPT
                         1452077       836,799  SONIC2.CPT
                         1452486       836,799  SONIC2B.CPT
                         1452895     1,405,280  SONIC3.CPT
                         1453582     2,995,531  SONIC3D.CPT
                         1455045     1,168,588  S_SPIN.CPT

Directory: MDROM_US
                         1455616     1,787,765  COMIX.CPT
                         1456489        46,934  FLICKY.CPT
                         1456512     1,886,551  KNU_P1.CPT
                         1457434     2,615,208  KNU_P2.CPT
                         1458711     2,866,160  KNU_P3.CPT
                         1460111       588,622  OOZE.CPT
                         1460399       641,540  PUYO_USA.CPT
                         1460713     1,575,127  RISTER.CPT
                         1461483     1,778,496  SANDK.CPT
                         1462352       423,591  SONIC1.CPT
                         1462559       425,672  SONIC1B.CPT
                         1462767       425,672  SONIC1U.CPT
                         1462975       836,799  SONIC2.CPT
                         1463384       836,799  SONIC2B.CPT
                         1463793     1,405,280  SONIC3.CPT
                         1464480     2,995,531  SONIC3D.CPT
                         1465943       585,865  S_SPIN.CPT

Directory: PAE_E
                         1466230        99,840  E3_MOVIE.PAE
                         1466279       379,556  E3_TITLE.PAE
                         1466465       758,225  EXTRA.PAE
                         1466836       144,088  MANUAL.PAE
                         1466907     1,058,817  MODE_GS.PAE
                         1467425       209,829  MODE_MC.PAE
                         1467528       137,125  MODE_MS.PAE
                         1467595       325,229  MODE_OPT.PAE
                         1467754       728,963  PAD.PAE
                         1468110       735,636  STATE.PAE
                         1468470       735,448  STATE_L.PAE

Directory: PAE_F

Directory: PAE_G

Directory: PAE_I

Directory: PAE_J
                         1468830       108,523  E3_MOVIE.PAE
                         1468883       381,837  E3_TITLE.PAE
                         1469070       762,750  EXTRA.PAE
                         1469443       144,844  MANUAL.PAE
                         1469514     1,042,956  MODE_GS.PAE
                         1470024       209,829  MODE_MC.PAE
                         1470127       125,010  MODE_MS.PAE
                         1470189       331,299  MODE_OPT.PAE
                         1470351       729,544  PAD.PAE
                         1470708       744,746  STATE.PAE
                         1471072       746,305  STATE_L.PAE

Directory: PAE_K

Directory: PAE_S

Directory: TEXT
                         1471437       266,456  TEXT_E.UTX
                         1471568        75,660  TEXT_F.UTX
                         1471605        75,328  TEXT_G.UTX
                         1471642        71,944  TEXT_I.UTX
                         1471678       141,292  TEXT_J.UTX
                         1471747        23,068  TEXT_K.UTX
                         1471759        71,604  TEXT_S.UTX

Directory: PAE_E_EU
                         1471794        99,840  E3_MOVIE.PAE
                         1471843       379,556  E3_TITLE.PAE
                         1472029       758,225  EXTRA.PAE
                         1472400       145,864  MANUAL.PAE
                         1472472     1,059,100  MODE_GS.PAE
                         1472990       209,829  MODE_MC.PAE
                         1473093       137,125  MODE_MS.PAE
                         1473160       325,229  MODE_OPT.PAE
                         1473319       728,964  PAD.PAE
                         1473675       736,169  STATE.PAE
                         1474035       735,980  STATE_L.PAE

Directory: COMIC
                         1474395         4,296  COMIC.XML
                         1474398         4,298  COMIC_E.XML
                         1474401    21,229,568  COM_1ST.AFS
                         1484767       673,792  COM_ABO.AFS
                         1485096       657,408  COM_ABOE.AFS
                         1485417     9,902,080  COM_K01.AFS
                         1490252    10,203,136  COM_K02.AFS
                         1495234    15,972,352  COM_K03.AFS
                         1503033    18,436,096  COM_O01.AFS
                         1512035    10,797,056  COM_S01.AFS
                         1517307    11,542,528  COM_S02.AFS
                         1522943    11,708,416  COM_S03.AFS
                         1528660    13,142,016  COM_S04.AFS
                         1535077    12,881,920  COM_S05.AFS
                         1541367    13,418,496  COM_S06.AFS
                         1547919    13,293,568  COM_S07.AFS
                         1554410    13,242,368  COM_S08.AFS
                         1560876    13,496,320  COM_S09.AFS
                         1567466    13,256,704  COM_S10.AFS
                         1573939    13,312,000  COM_S11.AFS
                         1580439     9,842,688  COM_S12.AFS
                         1585245    12,343,296  COM_S13.AFS
                         1591272     8,945,664  COM_S14.AFS
                         1595640    16,910,336  COM_SS01.AFS

Directory: COMMON
                         1603897           515  EMU_BG.PRS
                         1603898        14,004  LOGO_E.PRS
                         1603905        11,717  LOGO_J.PRS
                         1603911        41,440  MC_ICON.LST
                         1603932           964  MC_ICON.SYS
                         1603933        53,204  MC_ICO_G.LST
                         1603959        64,270  MC_ICO_M.LST
                         1603991       578,725  SEGA.PRS
                         1604274        83,571  STAFF.PRS
                         1604315        13,812  STAFF.XML
                         1604322        13,798  STAFF_EU.XML
                         1604329        81,833  STAFF_J.PRS
                         1604369        12,226  STAFF_J.XML

Directory: E3
                         1604375       283,385  E3_END.PAE
                         1604514     1,125,973  E3_MENU.PAE
                         1605064         7,746  E3_PAUSE.PAE
                         1605068       942,375  GG_MENU.PAE

Directory: GGROM
                         1605529       545,648  G_SONIC.CPT
                         1605796       243,716  LABYLIN.CPT
                         1605916       136,424  MBMACHIN.CPT
                         1605983       168,600  SONIC.CPT
                         1606066       210,129  SONICDRI.CPT
                         1606169       243,809  SONICTAI.CPT
                         1606289       244,442  SONIC_CH.CPT

Directory: ILLUST
                         1606409     3,338,240  CONTE_DK.AFS
                         1608039     4,370,432  CONTE_SH.AFS
                         1610173         2,750  ILLUST.XML
                         1610175     3,522,560  ILLUST01.AFS
                         1611895     3,319,808  ILLUST02.AFS
                         1613516     3,024,896  ILLUST03.AFS
                         1614993     2,011,136  ILLUST04.AFS
                         1615975     9,508,864  ILLUST05.AFS
                         1620618     3,977,216  ILLUST06.AFS
                         1622560     8,849,408  ILLUST07.AFS
                         1626881     3,086,336  ILLUST08.AFS

Directory: MANUAL
                         1628388     7,768,064  COMIX_M.AFS
                         1632181    14,110,720  DRIFT_G.AFS
                         1639071     1,222,656  FLICKY_M.AFS
                         1639668    15,366,144  G_SONIC.AFS
                         1647171    12,400,640  LABYLIN.AFS
                         1653226         7,344  MANUAL.XML
                         1653230     5,017,600  MEANBEAN.AFS
                         1655680     8,624,128  OOZ_M.AFS
                         1659891     5,185,536  PUYO_M.AFS
                         1662423     9,863,168  RISTAR_M.AFS
                         1667239    16,965,632  SANDK_M.AFS
                         1675523     6,881,280  SONIC1_G.AFS
                         1678883    27,813,888  SONIC2_M.AFS
                         1692464     3,004,416  SONIC3D.AFS
                         1693931    18,935,808  SONIC3_M.AFS
                         1703177    20,295,680  SONIC_M.AFS
                         1713087    14,088,192  S_SPIN.AFS
                         1719966    13,580,288  S_TAILS.AFS

Directory: MANUAL_E
                         1726597     6,162,432  COMIX_M.AFS
                         1729606    14,397,440  DRIFT_G.AFS
                         1736636     1,234,944  FLICKY_M.AFS
                         1737239     2,244,608  G_SONIC.AFS
                         1738335     2,592,768  LABYLIN.AFS
                         1739601         7,084  MANUAL.XML
                         1739605     4,278,272  MEANBEAN.AFS
                         1741694     3,756,032  OOZ_M.AFS
                         1743528     4,591,616  PUYO_M.AFS
                         1745770     3,022,848  RISTAR_M.AFS
                         1747246     4,464,640  SANDK_M.AFS
                         1749426     2,766,848  SONIC1_G.AFS
                         1750777     4,984,832  SONIC2_M.AFS
                         1753211     2,703,360  SONIC3D.AFS
                         1754531     5,730,304  SONIC3_M.AFS
                         1757329     3,407,872  SONIC_M.AFS
                         1758993     4,907,008  S_SPIN.AFS
                         1761389     4,100,096  S_TAILS.AFS

Directory: MDROM
                         1763391     1,792,692  COMIX.CPT
                         1764267        46,934  FLICKY.CPT
                         1764290     1,884,470  KNU_P1.CPT
                         1765211     2,615,208  KNU_P2.CPT
                         1766488     2,866,159  KNU_P3.CPT
                         1767888       588,622  OOZE.CPT
                         1768176       641,532  PUYO_USA.CPT
                         1768490     1,573,784  RISTER.CPT
                         1769259     1,778,496  SANDK.CPT
                         1770128       423,591  SONIC1.CPT
                         1770335       423,618  SONIC1B.CPT
                         1770542       425,672  SONIC1U.CPT
                         1770750       836,799  SONIC2.CPT
                         1771159       836,799  SONIC2B.CPT
                         1771568     1,405,279  SONIC3.CPT
                         1772255     2,995,531  SONIC3D.CPT
                         1773718       586,008  S_SPIN.CPT

Directory: MODE_GS
                         1774005     7,157,760  GAME00.SFD
                         1777500    15,441,920  GAME01.SFD
                         1785040    15,355,904  GAME02.SFD
                         1792538    17,620,992  GAME03.SFD
                         1801142    15,345,664  GAME04.SFD
                         1808635    15,151,104  GAME05.SFD
                         1816033    15,325,184  GAME06.SFD
                         1823516    15,988,736  GAME07.SFD
                         1831323    15,341,568  GAME08.SFD
                         1838814    15,327,232  GAME09.SFD
                         1846298    15,323,136  GAME10.SFD
                         1853780    15,341,568  GAME11.SFD
                         1861271    15,327,232  GAME12.SFD
                         1868755    15,329,280  GAME13.SFD
                         1876240    15,308,800  GAME14.SFD
                         1883715    15,323,136  GAME15.SFD
                         1891197    15,325,184  GAME16.SFD
                         1898680    15,341,568  GAME17.SFD
                         1906171    15,321,088  GAME18.SFD
                         1913652    15,351,808  GAME19.SFD
                         1921148    15,321,088  GAME20.SFD

Directory: SOUND
                         1928629     5,568,512  HISTORY.ADX
                         1931348     4,255,744  MANUAL.ADX
                         1933426     2,830,336  MENU.ADX
                         1934808       247,140  SE.ACX
                         1934929        88,762  SEGA.ADX
                         1934973     1,787,562  TITLE.ADX

Directory: MOVIE
                         1935846    17,803,264  EXTRA_A.SFD
                         1944539    17,803,264  EXTRA_B.SFD
                         1953232    67,555,328  EXTRA_D0.SFD
                         1986218    87,451,648  EXTRA_D1.SFD
                         2028919    48,242,688  EXTRA_O0.SFD
                         2052475   104,321,024  EXTRA_O1.SFD
                         2103413   103,268,352  EXTRA_O2.SFD
                         2153837    15,306,752  MANUAL_A.SFD
                         2161311    15,280,128  MANUAL_B.SFD
                         2168772       610,304  MODE_GS.SFD
                         2169070       610,304  MODE_GSE.SFD
                         2169368       612,352  MODE_GSU.SFD
                         2169667    22,851,584  MODE_MS.SFD
                         2180825    22,874,112  MODE_MSA.SFD
                         2191994    76,736,512  STAFF.SFD
                         2229463     8,239,104  TITLE.SFD
                         2233486   103,612,416  TITLE_LP.SFD
```

**Comment by claunia** [comment]
The `_DUMMY_.BIN` is bigger than 2GiB and that seems to be a limitation of 5.3. Will be fixed for 6.0.

**Comment by claunia** [comment]
Fixed in e34d0939a3effdc117ad91eab3719f127d987b65.
